src/main/java/edu/wpi/grip/ui/preview/ImageSocketPreviewView.java

```diff
@@ -56,62 +57,69 @@ public void onSocketChanged(SocketChangedEvent event) {
      * (Mat -> Frame -> BufferedImage -> JavaFX Image) and is way too slow to use for a real-time video.
      */
     private void convertImage() {
-        final Mat mat = this.getSocket().getValue();
-        final int width = mat.cols();
-        final int height = mat.rows();
-        final int channels = mat.channels();
-
-        assert channels == 3 || channels == 1 :
-                "Only 3-channel BGR images or single-channel grayscale images can be previewed";
-
-        assert mat.type() == CV_8U || mat.type() == CV_8S :
-                "Only images with 8 bits per channel can be previewed";
-
-        // Don't try to render empty images.
-        if (mat.empty()) {
-            this.imageView.setImage(null);
-            return;
-        }
-
-        // If the size of the Mat changed for whatever reason, allocate a new image with the proper dimensions and a buffer
-        // big enough to hold all of the pixels in the image.
-        if (this.image == null || this.image.getWidth() != width || this.image.getHeight() != height) {
-            this.image = new WritableImage(width, height);
-            this.pixels = IntBuffer.allocate(width * height);
-        }
+        synchronized (this) {
```

Comment:
Synchronized just means the the method won't be called reentrantly. This isn't really useful here because it's only ever called from one thread. The synchronized blocks are used to make sure this method and the lambda passed to `runLater` don't run at the same time, since they both access `this.pixels`.